packages/cw-umee-types/src/msg.rs

```diff
@@ -24,128 +22,112 @@ pub enum UmeeMsg {
 #[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
 #[serde(rename_all = "snake_case")]
 pub struct StructUmeeMsg {
-  assigned_msg: u16,
+  assigned_msg: MsgTypes,
   supply: Option<SupplyParams>,
   withdraw: Option<WithdrawParams>,
+  max_withdraw: Option<MsgMaxWithDrawParams>,
   collateralize: Option<CollateralizeParams>,
   decollateralize: Option<DecollateralizeParams>,
   borrow: Option<BorrowParams>,
   repay: Option<RepayParams>,
   liquidate: Option<LiquidateParams>,
+  supply_collateralize: Option<SupplyCollateralParams>,
 }
 
+fn default_struct_umee_msg(m: MsgTypes) -> StructUmeeMsg {
+  StructUmeeMsg {
+    assigned_msg: m,
+    supply: None,
+    withdraw: None,
+    collateralize: None,
+    decollateralize: None,
+    borrow: None,
+    repay: None,
+    liquidate: None,
+    max_withdraw: None,
+    supply_collateralize: None,
+  }
+}
 // Defines all the implementation related to the StructUmeeMsg
 // like creating new messages structs, it is needed because
 // the fields inside the struct are private, to avoid missmatching
 // the msg property with the assigned_msg field
 impl StructUmeeMsg {
   // valid returns true if is valid
   pub fn valid(&self) -> bool {
-    return self.assigned_msg > 0;
+    match &self.assigned_msg {
```

Comment:
This match will always resolve to the first case, because`self.assigned_msg` ahs `MsgTypes` always.